extensions/omni_schema/src/make_plan.sql

```diff
@@ -0,0 +1,129 @@
+    extension_record record;
+begin
+    perform dblink(current_db_conn, format('create database %I', plan_db));
+    perform dblink(plan_db_conn, 'create extension omni_schema cascade;');
+
+    -- prepare necessary extensions in the remote database
+    for extension_record in select distinct extension
+               from omni_schema.languages
+               union
+               select distinct file_processor_extension as extension
+               from omni_schema.languages
+               union
+               select distinct processor_extension as extension
+               from omni_schema.auxiliary_tools
+    loop
+            if exists (select from pg_extension where extname = extension_record.extension) then
+                perform dblink(plan_db_conn,
+                               format('create extension if not exists %s cascade', extension_record.extension));
+            end if;
+    end loop;
+
+    for migration in select * from dblink(
+        plan_db_conn,
+        format('select migration_filename, executed_statement, migration_statement, execution_error from omni_schema.assemble_schema(%L, omni_vfs.local_fs(%L))', plan_db_conn, path)
+    ) migrations (migration_filename text, executed_statement text, migration_statement text, execution_error text)
+    loop
+        -- first we have to handle errors coming from assemble_schema
+        if migration.executed_statement is not null and migration.execution_error is null then
+            begin
+                stmt := migration.executed_statement::omni_sql.statement;
+            exception
+                when others then
+                    get stacked diagnostics error_message = message_text,
+                                  hint_message = pg_exception_hint;
+                    raise exception 'Error "%" in %', error_message, migration.migration_filename
+                        using 
+                            detail = migration.executed_statement,
+                            hint = hint_message;
+            end;
+        elsif migration.execution_error is not null then
+            raise exception '% in %', migration.execution_error, migration.migration_filename
+                using 
+                    detail = migration.migration_statement;
+        end if;
+
+        -- now we branch between 2 execution modes: migration or source
+        if migration.migration_filename ~* 'migrations' then
+            -- migrations always execute only once
+            if not exists(select from omni_schema.migrations m where lower(m.name) = lower(omni_vfs.basename(migration.migration_filename))) 
+            then
+                return query select migration.migration_filename, migration.executed_statement::omni_sql.statement;
+            end if;
+        else
+            -- Non-migrations are picky, we error unless conditions are met
+            if not (
+                migration.executed_statement ~* 'select omni_python.install_requirements'
```

Comment:
is this particular line fragile? or good enough for now?

Reply:
Yes, I was not happy with this. Was mostly a stop-gap solution. But we should review the approach to `omni_python` and other language loaders. Perhaps they should live in their own modules and we could invert the dependency, so they would use `omni_schema` to build a more refined plan.